libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -81,18 +81,17 @@
     }
 
     private interface UnknownFieldParser<Value, Context> {
-
-        void acceptUnknownField(String parserName, String field, XContentLocation location, XContentParser parser,
-                                Value value, Context context) throws IOException;
+        void acceptUnknownField(ObjectParser<Value, Context> objectParser, String field, XContentLocation location, XContentParser parser,
+                Value value, Context context) throws IOException;
```

Comment:
I *think* passing `ObjectParser` here is ok because the interface is entirely private already. I could certainly be convinced otherwise though.